apps/easypid/src/features/scan/FunkeQrScannerScreen.tsx

```diff
@@ -0,0 +1,283 @@
+import { QrScanner } from '@package/scanner'
+import {
+  AnimatedStack,
+  Heading,
+  HeroIcons,
+  Loader,
+  Page,
+  Paragraph,
+  Spinner,
+  Stack,
+  useSpringify,
+  useToastController,
+} from '@package/ui'
+import { useIsFocused } from '@react-navigation/native'
+import { useRouter } from 'expo-router'
+import React, { useEffect, useState } from 'react'
+import QRCode from 'react-native-qrcode-svg'
+
+import { type CredentialDataHandlerOptions, isAndroid, useCredentialDataHandler, useHaptics } from '@package/app'
+import { Alert, Linking, Platform, useWindowDimensions } from 'react-native'
+import { FadeIn, FadeOut, LinearTransition, useAnimatedStyle, withTiming } from 'react-native-reanimated'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+
+import { pidSchemes } from '@easypid/constants'
+import easypidLogo from '../../../assets/easypid.png'
+import { checkMdocPermissions, getMdocQrCode, requestMdocPermissions, waitForDeviceRequest } from '../proximity'
+
+const unsupportedUrlPrefixes = ['_oob=']
+
+interface QrScannerScreenProps {
+  credentialDataHandlerOptions?: CredentialDataHandlerOptions
+}
+
+export function FunkeQrScannerScreen({ credentialDataHandlerOptions }: QrScannerScreenProps) {
```

Comment:
it's sort of ok with the deadline now, but i think it would have been better to split this into a generic scan/show component that can be used in both paradym and easypid and then add a `showQrCode` and exrract all the mdoc specific logic out of here.